### DISCUSSION TITLE
Recomended resources for running this?

### DISCUSSION THREAD

**Discussion by Lexicality** [pypiserver/pypiserver#522] [Q&A] [answered]
Is there any guidance for the kind of resources (CPU/Memory) I should provide to a VM running the docker image? 

I assume the requirements scale with usage but how? Would going from 10 users to 100 require 10 times the resources as before?

Thanks

**Comment by dee-me-tree-or-love** [comment]
Hey @Lexicality, thanks a lot for opening this question! That's a very good discussion item 👍

To my knowledge, we didn't come up with a good recommendation/guidance on resource usage 🙈 So in fact, this looks like a perfect opportunity to get some community-based insights 🤔😊    

---

To be honest, I've personally used `pypiserver` mostly for low-traffic use cases (e.g. local experimentation platforms). These had minimal-to-moderate traffic and I had a good experience running the setup on sub **1CPU and approx. 256MiB memory** ([configured in Kubernetes](https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/memory-constraint-namespace/) & [observed](https://cloud.google.com/monitoring/api/metrics_gcp#gcp-appengine:~:text=active%2C%20idle%2C%20etc).-,system/memory/usage,-GA%0AMemory%20usage) in [AppEngine](https://cloud.google.com/appengine/docs/flexible/python/runtime) sometime). Also, I came across a [nice blog post](https://testdriven.io/blog/private-pypi/) was written by @mjhea0 who makes use of `EC2-micro` with 1CPU and 1GB and that also seems to be working quite well.

> I assume the requirements scale with usage but how? Would going from 10 users to 100 require 10 times the resources as before?

My intuition would be that it depends on quite a few factors: at least caching, HTTP server backend used. We're using `bottle` as our HTTP framework under the hood, so `pypi-server` borrows some of its behaviour in terms of scaling too. Generally, I wouldn't expect such a high coefficient of resource use, but wouldn't be able to point to the good one on top of my head 😁 But I'll try to dig deeper into it and come back with findings here. 

**As a quick start idea**, I'd personally go for a small resource setup and then monitor the resource usage to see what tweaks are possible. What do you think @Lexicality?

---

In fact, that would be awesome to make this base resource usage recommendation available but it might take a bit of time. I'd be very happy to hear about any other experiences if anyone wants to share too. 🍀 

I hope this somewhat helps, I'm sorry I can't provide a more accurate suggestion yet.

**Reply by Lexicality** [reply]
Thank you! I'm still in the fact-finding stage but I'll report back if I have useful results.

**Comment by jayeff** [comment, marked as answer]
Our pypiserver instance runs on 1vCPU and 1GB RAM which is plenty for our use case (internal usage, a couple of packages, tens of users). I would assume this setup scales up to hunderds of users before we maybe run into performance problems 🤔

If you have a bigger use case I would recommend to do some load testing and if you do, please share your results
